### ISSUE TITLE
Traceplot parity

### ISSUE THREAD

**Issue by ColCarroll** [arviz-devs/arviz#94]
Is it intentional to have a different traceplot from pymc3?  Each element of an RV with shape > 1 gets a plot in `arviz`, while they are all the same for `pymc3`.  

Explicitly, with eight schools (I only capture the first few plots from arviz):

```
J = 8
y = np.array([28,  8, -3,  7, -1,  1, 18, 12])
sigma = np.array([15, 10, 16, 11,  9, 11, 10, 18])

with pm.Model() as hierarchical:
    eta = pm.Normal('eta', 0, 1, shape=J)
    mu = pm.Normal('mu', 0, sd=1e6)
    tau = pm.HalfCauchy('tau', 5)
    theta = pm.Deterministic('theta', mu + tau*eta)
    obs = pm.Normal('obs', theta, sd=sigma, observed=y)
    trace_h = pm.sample(1000)
```

![image](https://user-images.githubusercontent.com/2295568/40398462-3c442368-5e06-11e8-8020-337346b185e9.png)

**Comment by aloctavodia** [comment]
Yes, this is intentional for traceplot and other plots.

**Comment by ColCarroll** [comment]
Thanks!

**Comment by junpenglao** [comment]
Is there a plan to add an option to have the old pymc3 style plot? 

Otherwise it might be useful to restrict the number of plot traceplot (or any other plots, for that matter) products. I am re-running some old codes and if there is a lot of dimension it will break arviz (since it is flattening the dimensions)

**Comment by aloctavodia** [comment]
Do you mean to limit, by default, the output to a given number of plots? 

You can use the variable names and coordinates to, explicitly, reduce the number of plots, do you think this is not enough?

**Comment by ColCarroll** [comment]
I could imagine doing this at the pymc3 level (you can do fancy things with the axes to "overplot"), so that `pm.traceplot` looks the same, or adding an option in `arviz` like (somewhat tongue-in-cheek) `style={'modern'|'classic'}`, or (more reasonable) `style={'facet'|'compact'}`.

**Comment by ahartikainen** [comment]
This is just my opinion/question: How often all the components in one parameter describe the same thing?

What could be the future, is to create a book for all the plots or create interactive visualization where user can select the number of "rows" and "cols" for number of plots and also if parameters are combined or not (even on parameter level; e.g. select only theta).

But book would be cool.

**Comment by junpenglao** [comment]
> Do you mean to limit, by default, the output to a given number of plots?

I think having some default is good - otherwise if you just naively call trace_plot with large number of plot it might hang for a while and eventually output an error.

**Comment by ColCarroll** [comment]
Rebumping this after it came up in PyMC3: I think there is interest in 
1. Global config limit on the number of axes arviz will create (say, 50?)
2. A `kind={'facet' | 'compact'}` option for traceplot. See [cell 23](https://docs.pymc.io/notebooks/bayesian_neural_network_advi.html) in this notebook for an example that could reasonably call for a `'compact'` option.

**Comment by michaelosthege** [comment]
One difficulty with combined plots (I prefer them in most cases) is the overlay of the chains on the right. Maybe, they should get `alpha=1/ndim` or so.

**Comment by twiecki** [comment]
Advocating for returning to the pymc3 way by default. It is much worse UX to take forever and then break, rather than displaying a crowded plot which might trigger a user to look at the docs to find that option.

**Comment by junpenglao** [comment]
Or having an option to do pymc3 type of plot so we can use that style in pymc3.
In general, I think it is better style in Arviz as overlaying lots of plots does not really makes sense, but myself also prefer to lazily just plot everything so I can at least eyeball the obviously wrong quickly.

**Comment by twiecki** [comment]
I would argue that even for arviz it doesn't make sense. Usually if a plot is too crowded it will also be too large to be plotted flattened. One results in a crowded plot the other in a hanging kernel.

**Comment by aloctavodia** [comment]
This have been solved by  #679
